docs/content/docs/sql-api/querying-tables.md

```diff
@@ -24,99 +24,98 @@ specific language governing permissions and limitations
-SET 'execution.runtime-mode' = 'batch';
-SELECT * FROM MyTable;
-```
-
-## Query Engines
-
-Table Store not only supports Flink SQL queries natively but also provides
-queries from other popular engines. See [Engines]({{< ref "docs/engines/overview" >}})
-
-## Query Optimization
-
-It is highly recommended to specify partition and primary key filters
-along with the query, which will speed up the data skipping of the query.
-
-The filter functions that can accelerate data skipping are:
-- `=`
-- `<`
-- `<=`
-- `>`
-- `>=`
-- `IN (...)`
-- `LIKE 'abc%'`
-- `IS NULL`
-
-Table Store will sort the data by primary key, which speeds up the point queries
-and range queries. When using a composite primary key, it is best for the query
-filters to form a [leftmost prefix](https://dev.mysql.com/doc/refman/5.7/en/multiple-column-indexes.html)
-of the primary key for good acceleration.
-
-Suppose that a table has the following specification:
-
-```sql
-CREATE TABLE orders (
-    catalog_id BIGINT,
-    order_id BIGINT,
-    .....,
-    PRIMARY KEY (catalog_id, order_id) NOT ENFORCED -- composite primary key
-)
-```
-
-The query obtains a good acceleration by specifying a range filter for
-the leftmost prefix of the primary key.
-
-```sql
-SELECT * FROM orders WHERE catalog_id=1025;
-
-SELECT * FROM orders WHERE catalog_id=1025 AND order_id=29495;
-
-SELECT * FROM orders
-  WHERE catalog_id=1025
-  AND order_id>2035 AND order_id<6000;
-```
-
-However, the following filter cannot accelerate the query well.
-
-```sql
-SELECT * FROM orders WHERE order_id=29495;
-
-SELECT * FROM orders WHERE catalog_id=1025 OR order_id=29495;
-```
-
-## Snapshots Table
-
-You can query the snapshot history information of the table through Flink SQL.
+# Querying Tables
+
+Just like all other tables, Table Store tables can be queried with `SELECT` statement.
+
+## Scan Mode
+
+By specifying the `scan.mode` table property, users can specify where and how Table Store sources should produce records.
+
+<table class="table table-bordered">
+<thead>
+<tr>
+<th>Scan Mode</th>
+<th>Batch Source Behavior</th>
+<th>Streaming Source Behavior</th>
+</tr>
+</thead>
+<tbody>
+<tr>
+<td>default</td>
+<td colspan="2">
+The default scan mode. Determines actual scan mode according to other table properties. If "scan.timestamp-millis" is set the actual scan mode will be "from-timestamp". Otherwise the actual scan mode will be "latest-full".
+</td>
+</tr>
+<tr>
+<td>latest-full</td>
+<td>
+Produces the latest snapshot of table.
+</td>
+<td>
+Produces the latest snapshot on the table upon first startup, and continues to read the following changes.
+</td>
+</tr>
+<tr>
+<td>compacted-full</td>
+<td>
+Produces the snapshot after the latest <a href="{{< ref "docs/concepts/lsm-trees#compactions" >}}">compaction</a>.
+</td>
+<td>
+Produces the snapshot after the latest compaction on the table upon first startup, and continues to read the following changes.
+</td>
+</tr>
+<tr>
+<td>latest</td>
```

Comment:
We can remove this legacy one

Reply:
`latest` is not the legacy one. That's `full`.